### DISCUSSION TITLE
Provide password for database connection when importing

### DISCUSSION THREAD

**Discussion by Papipo** [apache/superset#18536] [Ideas]
**Is your feature request related to a problem? Please describe.**
Passwords are masked in exports, thus they can't be imported back.

**Describe the solution you'd like**
Be able to provide a flag to the `import-directory` command to specify such password and be able to create/update the connection.

**Additional context**
Convo with @betodealmeida: 
https://apache-superset.slack.com/archives/C016B3LG5B4/p1638755013257400?thread_ts=1638624207.253200&cid=C016B3LG5B4

**Comment by Papipo** [comment]
Is this planned to be tackled any time soon?

**Comment by betodealmeida** [comment]
@Papipo I'm not aware of anyone who's planning to work on this. At my job I wrote a CLI that allows people to manage their assets as code. The tool is free to use (but not technically open-source) and works with Superset, and when you try to import an asset with a masked password it will prompt the user for it.

You can read more about it here: https://preset.io/blog/version-control-superset-charts-dashboards-superset/

The CLI lives here: https://github.com/preset-io/backend-sdk

The logic for asking for the password lives here: https://github.com/preset-io/backend-sdk/blob/d05d01b6a256ca925f7141c7d1688d80681de6d9/src/preset_cli/cli/superset/sync/native/command.py#L311-L322

If you want to try to add that logic to the Superset import (probably to `superset.examples.utils.load_configs_from_directory`) I can help you with code review.
